lisa/sut_orchestrator/azure/common.py

```diff
@@ -590,6 +590,19 @@ def check_or_create_resource_group(
                     resource_group_name, {"location": location}
                 )
 
+            timeout = 30
+            timeout_timer = create_timer()
+            while timeout_timer.elapsed(False) < timeout:
+                az_shared_rg_exists = rm_client.resource_groups.check_existence(
```

Comment:
Is it fixed by wait longer? The create_or_update is a synchronized API as before.